pkg/app/validate_app.go

```diff
@@ -0,0 +1,67 @@
+package app
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/v3/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/k8sclient/v5/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"k8s.io/api/admission/v1beta1"
+
+	"github.com/giantswarm/app-admission-controller/config"
+	"github.com/giantswarm/app-admission-controller/pkg/validator"
+)
+
+const (
+	Name = "app"
+)
+
+type Validator struct {
+	k8sClient k8sclient.Interface
+	logger    micrologger.Logger
+}
+
+func NewValidator(config config.Config) (*Validator, error) {
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	validator := &Validator{
+		k8sClient: config.K8sClient,
+		logger:    config.Logger,
+	}
+
+	return validator, nil
+}
+
+func (v *Validator) Log(keyVals ...interface{}) {
+	v.logger.Log(keyVals...)
+}
+
+func (v *Validator) Resource() string {
+	return Name
+}
+
+func (v *Validator) Validate(request *v1beta1.AdmissionRequest) (bool, error) {
+	ctx := context.Background()
+
+	var app v1alpha1.App
+
+	if _, _, err := validator.Deserializer.Decode(request.Object.Raw, nil, &app); err != nil {
+		return false, microerror.Maskf(parsingFailedError, "unable to parse app: %#v", err)
+	}
+
+	appAllowed, err := v.ValidateApp(ctx, app)
+	if err != nil {
+		return false, microerror.Mask(err)
+	}
+
+	return appAllowed, nil
+}
+
+func (v *Validator) ValidateApp(ctx context.Context, app v1alpha1.App) (bool, error) {
+	return true, nil
```

Comment:
Validation logic will be moved to the app library. So it can be reused with the validation resource in app-operator.